WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -32,6 +33,27 @@ public CoinJoinManager(WalletManager walletManager, RoundStateUpdater roundStatu
 		public IWabiSabiApiRequestHandler ArenaRequestHandler { get; }
 		public RoundStateUpdater RoundStatusUpdater { get; }
 		public ServiceConfiguration ServiceConfiguration { get; }
+		private ImmutableDictionary<string, WalletTrackingData> TrackedWallets { get; set; } = ImmutableDictionary<string, WalletTrackingData>.Empty;
+
+		public CoinJoinClientState HighestCoinJoinClientState
+		{
+			get
+			{
+				var coinjoinClients = TrackedWallets.Values
+					.Where(wtd => !wtd.CoinJoinTask.IsCompleted)
+					.GroupBy(wtd => wtd.CoinJoinClient.State)
+					.ToImmutableArray();
+
+				if (coinjoinClients.Any(grp => grp.Key is CoinJoinClientState.InCriticalPhase))
+				{
+					return CoinJoinClientState.InCriticalPhase;
+				}
+
+				return coinjoinClients.Any(grp => grp.Key is CoinJoinClientState.InProgress)
+					? CoinJoinClientState.InProgress
+					: CoinJoinClientState.Idle;
```

Comment:
Isn't the following code more simple?

```suggestion
				var coinjoinClients = TrackedWallets.Values
					.Where(wtd => !wtd.CoinJoinTask.IsCompleted)
					.Select(wtd => wtd.CoinJoinClient.State)
					.ToImmutableArray();

				if (coinjoinClients.Contains(CoinJoinClientState.InCriticalPhase))
				{
					return CoinJoinClientState.InCriticalPhase;
				}

				return coinjoinClients.Contains(CoinJoinClientState.InProgress)
					? CoinJoinClientState.InProgress
					: CoinJoinClientState.Idle;
```

Reply:
Refactored this part according to Lucas's comment.